src/foam/nanos/auth/CapabilityAuthService.js

```diff
@@ -248,17 +203,16 @@ foam.CLASS({
             }
 
             // Add the result to the cache
+            (( Map<String, Boolean> ) getCache()).put(agentKey, result);
             if ( result ) {
-              if ( agentKey != null ) (( Map<String, Boolean> ) getCache()).put(agentKey, result);
               return true;
             }
           }
-        } catch (Exception e) {
+        } catch ( Exception e ) {
           Logger logger = (Logger) x.get("logger");
           logger.error("check", permission, e);
```

Comment:
While here, can you consider changing 'check' to 'capabilityCheck'. I'm thinking e will give us stack trace, but 'check' seems useless. This function can get called from either auth.check or auth.checkUser